### PR TITLE
Io 4 dev Macro

### PR DESCRIPTION
**Pull request by Lefiuu**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        this.reverseDriver();
+    }
+    public void setLineDriver(Job2dDriver jd){
+        this.jd=jd;
+    }
+    public void reverseDriver(){
+        DriverFeature.getDriverManager().setCurrentDriver(jd);
+    }
+    @Override
+    public void setPosition(int i, int i1) {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java:32]: change name of params to x and y

File: src/main/java/edu/kis/powp/jobs2d/features/MacroFeature.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import edu.kis.powp.jobs2d.command.manager.LoggerCommandChangeObserver;
+import edu.kis.powp.jobs2d.drivers.MacroDriver;
+
+public class MacroFeature {
+
+    private static MacroDriver macroDriver= new MacroDriver();
+
+    public static void setupMacroManager() {
+
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/MacroFeature.java:12]: unnecessary empty lines

File: src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+import edu.kis.powp.jobs2d.features.DriverFeature;
+
+
+/**
+ * Driver command Manager.
+ */
+public class MacroDriver implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java:17]: The name of the design pattern used should be added (in this case)

File: src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    private Job2dDriver jd;
+    public void clearMacro(){
+        commands.clear();
+        this.reverseDriver();
+    }
+    public void setLineDriver(Job2dDriver jd){
+        this.jd=jd;
+    }
+    public void reverseDriver(){
+        DriverFeature.getDriverManager().setCurrentDriver(jd);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java:29]: Please try to remove the dependency to DriverFeature (without adding new ones).

**Review by kowallus — CHANGES_REQUESTED** [review]
After cleanup I see some new problems.

**Comment by kowallus** [comment]
Reopen PR after modifications. Pull changes from the main branch!

File: src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java (hunk 3 above)
Comment (Lefiuu) [reply src/main/java/edu/kis/powp/jobs2d/drivers/MacroDriver.java:17]: How should this be written?

**Pull request closed without merging**